django/utils/numberformat.py

```diff
@@ -18,38 +18,58 @@ def format(number, decimal_sep, decimal_pos=None, grouping=0, thousand_sep='',
     * grouping: Number of digits in every group limited by thousand separator
     * thousand_sep: Thousand separator symbol (for example ",")
     """
-    use_grouping = settings.USE_L10N and settings.USE_THOUSAND_SEPARATOR
-    use_grouping = use_grouping or force_grouping
-    use_grouping = use_grouping and grouping > 0
-    # Make the common case fast
-    if isinstance(number, int) and not use_grouping and not decimal_pos:
-        return mark_safe(six.text_type(number))
-    # sign
-    sign = ''
+    use_grouping = grouping > 0 and (force_grouping or
+                                     (settings.USE_THOUSAND_SEPARATOR and settings.USE_L10N))
+    if not use_grouping or grouping == 3:
+        try:
```

Comment:
Do we need such a large try/except block? That makes it difficult to spot what statement(s) might throw an exception and could hide other bugs.